### ISSUE TITLE
Add a rule to avoid default case in switch expression

### ISSUE THREAD

**Issue by NolwennD** [checkstyle/checkstyle#13865]
Exhaustiveness check by `javac` in switch expression is a key feature since java 17 for enum and for sealed type since java 21. If we add a new type in hierarchy and there is a default case it will be fine for `javac` but the new type is not managed. Clearly default case in switch expression is a code smell most of the time.

It exists, more or less, the opposite rule [MissingSwitchDefault](https://checkstyle.org/checks/coding/missingswitchdefault.html) 

```java
public sealed interface Bob {
	record A() implements Bob {}
	record B() implements Bob {}
	record C() implements Bob {}

	static String goodSwitchExpression(Bob bob) {
	  return switch (bob) {
	    case A __ -> "a"; // in java 22 with unamed pattern it will be less ugly
	    case B __ -> "b";
	    case C __ -> "c";
          };
	}

	static String badSwitchExpression(Bob bob) {
	  return switch (bob) {
	    case A __ -> "a";
	    case B __ -> "b";
	    default -> "default";
          };
	}
}
```

**Comment by romani** [comment]
Please strictly follow issue template it will let you share all details that we need for issue approval.

**Comment by nrmancuso** [comment]
We will not be able to support exhaustive switch semantics, Checkstyle is not type aware. Please see https://checkstyle.sourceforge.io/writingchecks.html#Limitations for more information.

**Comment by NolwennD** [comment]
Thank you for explanation, I miss this limitation. Do you known a tool to check this kind of rule ? Spotbugs maybe ?

**Comment by nrmancuso** [comment]
@NolwennD I am not aware of any tools that do this, maybe try IDEA inspections?

**Comment by NolwennD** [comment]
Thank you @nrmancuso . I close the issue.
